### PR TITLE
Final touch

### PR DESCRIPTION
**Pull request by AsumVictor**
# Final touch
In this PR, we have finished implementing all features and functionality for the movie app fully working.

## Type of Changes
- A home page showing a list of movies that you can "like."
- A popup window with more data about a movie that you can use to comment on it.
- When the user clicks on the Like button of a Movie, the interaction is recorded in the Involvement API and the screen is updated.
- When the user clicks on the "Comments" button, the Comments popup appears.

## Checklist
 ✔Our code follows the style guidelines of this project
 ✔ Our have performed a self-review of my code
 ✔There is no linter error in my code

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -61,231 +33,158 @@ For more information on the importance of a professional README for your reposit
...
-<!-- LICENSE -->
 
 ## 📝 License <a name="license"></a>
 
-This project is [MIT](./LICENSE) licensed.
-
-_NOTE: we recommend using the [MIT license](https://choosealicense.com/licenses/mit/) - you can set it up quickly by [using templates available on GitHub](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/adding-a-license-to-a-repository). You can also use [any other license](https://choosealicense.com/licenses/) if you wish._
-
-<p align="right">(<a href="#readme-top">back to top</a>)</p>
+This project is [MIT](./LICENSE) licensed.
```
Comment (jerryowusu) [inline README.md]: The license file is not working and leads to an error page (404 page not found). Kindly correct it to give your project a more professional look. :wink:

File: README.md [hunk 2]
```diff
@@ -61,231 +33,158 @@ For more information on the importance of a professional README for your reposit
...
-- **[key_feature_3]**
 
 <p align="right">(<a href="#readme-top">back to top</a>)</p>
 
-<!-- LIVE DEMO -->
 
 ## 🚀 Live Demo <a name="live-demo"></a>
 
-> Add a link to your deployed project.
+- [Live Demo]()
```
Comment (jerryowusu) [inline README.md]: Kindly provide a link/url to the live demo link so that it can lead to the right page when you click on it. This will give your project a more professional outlook and up to standard.

File: Testing/likeCounter.test.js [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+    selectElFromDom.mockReturnValueOnce(mockLikeSpanCounter).mockReturnValueOnce(mockLoveBtn);
+
+    await likeCount(null, '123');
+
+    expect(selectElFromDom).toHaveBeenCalledWith('.like-counter123');
+    expect(selectElFromDom).toHaveBeenCalledWith('#love123');
+    expect(mockLoveBtn.classList.contains).toHaveBeenCalledWith('fa-solid');
+    expect(mockLoveBtn.classList.replace).toHaveBeenCalledWith('fa-regular', 'fa-solid');
+    expect(mockLikeSpanCounter.innerHTML).toBe('4');
+  });
```
Comment (jerryowusu) [inline Testing/likeCounter.test.js]: Nice work so far. However, the counter functions should be covered for all edge cases. For example, you should think of at least a scenario in which there are 0 items on the page (the counter function should return 0). Kindly write a test for that.

File: src/modules/apis.js [hunk 4]
```diff
@@ -0,0 +1,99 @@
+import { MOVIE_API_URL, BASE_URL, movieContainer } from './variables.js';
+
+/* eslint no-underscore-dangle: ["error", {"allow": ["_embedded"]}] */
```
Comment (jerryowusu) [inline src/modules/apis.js]: Kindly do not disable the linters since they serve as valuable tools to maintain code quality, catch potential bugs, enforce best practices, and improve maintainability. I believe you can work around this one without disabling it. Therefore, kindly consider removing this line and ensure to work within the confines of the linters. :pray:

**Review by jerryowusu — CHANGES_REQUESTED** [review]
Hi @AsumVictor  and @Dorcas126 

Good job so far!
Your project is looking fantastic so far :star_struck: 

However, there are few changes you need to make in order to move on to the next level :sunglasses:

almost

## Highlights
Nice design :+1: 
Neat and well organized code :+1: 

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 1 above)
Comment (AsumVictor) [reply README.md]: Thank you. You have included the license file.

File: README.md (hunk 2 above)
Comment (AsumVictor) [reply README.md]: Great!

File: Testing/likeCounter.test.js (hunk 3 above)
Comment (AsumVictor) [reply Testing/likeCounter.test.js]: Thank you! We have included a test for no item on the page

File: src/modules/apis.js (hunk 4 above)
Comment (AsumVictor) [reply src/modules/apis.js]: We have resolved it. Thank you

File: README.md [hunk 5]
```diff
@@ -61,231 +33,160 @@ For more information on the importance of a professional README for your reposit
...
-<!-- LICENSE -->
 
 ## 📝 License <a name="license"></a>
 
-This project is [MIT](./LICENSE) licensed.
-
-_NOTE: we recommend using the [MIT license](https://choosealicense.com/licenses/mit/) - you can set it up quickly by [using templates available on GitHub](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/adding-a-license-to-a-repository). You can also use [any other license](https://choosealicense.com/licenses/) if you wish._
-
-<p align="right">(<a href="#readme-top">back to top</a>)</p>
+This project is [MIT](./LICENSE) licensed.
```
Comment (iambenkis) [inline README.md]: - [x] Kindly fix the MIT license link , because it still hold a broken link 😄 

Capture

File: src/template.html [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+    <nav class="nav">
+      <div class="logo-movie">
+        <a href="#" class="logo">
+          <span>D</span>
+          oc
+          <span>V</span>
+          ic
+        </a>
+        <span>movies</span>
+      </div>
```
Comment (iambenkis) [inline src/template.html:30]: - [x] [**OPTIONAL**] I noticed that your header does not have the number of all the movies , kindly ensure you add it to get the job done 👍 

Capture


Capture

**Review by iambenkis — CHANGES_REQUESTED** [review]
#### Hi @AsumVictor , @jerryowusu 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!
You did a great job, keep up the good work 👍🏼

giphy


## Highlights

✔️  No linter errors
✔️ Used Gitflow  

## Suggested changes
_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

File: src/template.html (hunk 6 above)
Comment (AbayomiOlaoye) [reply src/template.html:30]: - [ ] [OPTIONAL] I think it will be cool you implement this kind suggestion for improvement by the reviewer :kneeling_person: What you have here is a counter for another project - spaceship, not for this movie web app.

**Review by AbayomiOlaoye — APPROVED** [review]
## STATUS: APPROVED :green_circle:  

Hi @Team  :wave: :wave:, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

applause

## Highlights
- Cleanly written code :white_check_mark: 
- App runs well and functions as expected ✅ 
- Nicely implemented UI styles ✅ 
- Great teamwork ✅ 
- Comments and likes are created efficiently :heavy_check_mark: 
- Fetches API correctly :heavy_check_mark: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**Comment by AsumVictor** [comment]
Thank you all for your help. We really appreciate your effort.
